### ISSUE TITLE
docker container start failed and 100% disk usage

### ISSUE THREAD

**Issue by xiemeilong** [scylladb/scylla#5173]
*Installation details*
Scylla version (or git commit hash): 3.x.x  2.6.3  (what I tried)
Cluster size: 1
OS (RHEL/CentOS/Ubuntu/AWS AMI): docker on debian9

Platform (physical/VM/cloud instance type/docker): docker 
Disks: (SSD/HDD, count)  1.7T HDD

I just run a container on debian 9,   100% disk usage after start failed

```
docker run --name some-scylla --hostname some-scylla -d scylladb/scylla
```

log
```
2019-10-14 10:02:16,373 INFO success: node-exporter entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2019-10-14 10:02:16,373 INFO success: sshd entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2019-10-14 10:02:16,373 INFO success: scylla-housekeeping entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
Traceback (most recent call last):
  File "/opt/scylladb/python3/lib64/python3.7/urllib/request.py", line 1317, in do_open
    encode_chunked=req.has_header('Transfer-encoding'))
  File "/opt/scylladb/python3/lib64/python3.7/http/client.py", line 1229, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/opt/scylladb/python3/lib64/python3.7/http/client.py", line 1275, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/opt/scylladb/python3/lib64/python3.7/http/client.py", line 1224, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/opt/scylladb/python3/lib64/python3.7/http/client.py", line 1016, in _send_output
    self.send(msg)
  File "/opt/scylladb/python3/lib64/python3.7/http/client.py", line 956, in send
    self.connect()
  File "/opt/scylladb/python3/lib64/python3.7/http/client.py", line 928, in connect
    (self.host,self.port), self.timeout, self.source_address)
  File "/opt/scylladb/python3/lib64/python3.7/socket.py", line 727, in create_connection
    raise err
  File "/opt/scylladb/python3/lib64/python3.7/socket.py", line 716, in create_connection
    sock.connect(sa)
OSError: [Errno 99] Cannot assign requested address

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/scylla/libexec/scylla-housekeeping", line 184, in <module>
    args.func(args)
  File "/usr/lib/scylla/libexec/scylla-housekeeping", line 110, in check_version
    current_version = sanitize_version(get_api('/storage_service/scylla_release_version'))
  File "/usr/lib/scylla/libexec/scylla-housekeeping", line 69, in get_api
    return get_json_from_url("http://" + api_address + path)
  File "/usr/lib/scylla/libexec/scylla-housekeeping", line 64, in get_json_from_url
    data = urllib.request.urlopen(path).read()
  File "/opt/scylladb/python3/lib64/python3.7/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/opt/scylladb/python3/lib64/python3.7/urllib/request.py", line 525, in open
    response = self._open(req, data)
  File "/opt/scylladb/python3/lib64/python3.7/urllib/request.py", line 543, in _open
    '_open', req)
  File "/opt/scylladb/python3/lib64/python3.7/urllib/request.py", line 503, in _call_chain
    result = func(*args)
  File "/opt/scylladb/python3/lib64/python3.7/urllib/request.py", line 1345, in http_open
    return self.do_open(http.client.HTTPConnection, req)
  File "/opt/scylladb/python3/lib64/python3.7/urllib/request.py", line 1319, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [Errno 99] Cannot assign requested address>
2019-10-14 10:02:28,879 INFO exited: scylla (terminated by SIGABRT (core dumped); not expected)
2019-10-14 10:02:29,881 INFO spawned: 'scylla' with pid 200
scylla: ../../src/core/reactor.cc:1297: seastar::reactor::reactor(unsigned int, seastar::reactor_backend_selector, seastar::reactor_config): Assertion `r >= 0' failed.
Aborting on shard 32.
Backtrace:
  0x00000000041a0bf2
  0x000000000408df05
  0x000000000408e205
  0x000000000408e2b3
  0x00007fddb04b102f
  /opt/scylladb/libreloc/libc.so.6+0x000000000003853e
  /opt/scylladb/libreloc/libc.so.6+0x0000000000022894
  /opt/scylladb/libreloc/libc.so.6+0x0000000000022768
  /opt/scylladb/libreloc/libc.so.6+0x00000000000309f5
  0x0000000004196ac6
  0x0000000004196db3
  0x000000000419705a
  0x00000000040596ad
  /opt/scylladb/libreloc/libpthread.so.0+0x000000000000858d
  /opt/scylladb/libreloc/libc.so.6+0x00000000000fd6a2
WARN  2019-10-14 10:02:30,456 [shard 0] seastar - Unable to set SCHED_FIFO scheduling policy for timer thread; latency impact possible. Try adding CAP_SYS_NICE
scylla: ../../src/core/reactor.cc:1297: seastar::reactor::reactor(unsigned int, seastar::reactor_backend_selector, seastar::reactor_config): Assertion `r >= 0' failed.
scylla: ../../src/core/reactor.cc:1297: seastar::reactor::reactor(unsigned int, seastar::reactor_backend_selector, seastar::reactor_config): Assertion `r >= 0' failed.
scylla: ../../src/core/reactor.cc:1297: seastar::reactor::reactor(unsigned int, seastar::reactor_backend_selector, seastar::reactor_config): Assertion `r >= 0' failed.
2019-10-14 10:02:31,459 INFO success: scylla entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2019-10-14 10:02:53,628 INFO exited: scylla (terminated by SIGABRT (core dumped); not expected)
2019-10-14 10:02:54,630 INFO spawned: 'scylla' with pid 335
WARN  2019-10-14 10:02:55,187 [shard 0] seastar - Unable to set SCHED_FIFO scheduling policy for timer thread; latency impact possible. Try adding CAP_SYS_NICE
scylla: ../../src/core/reactor.cc:1297: seastar::reactor::reactor(unsigned int, seastar::reactor_backend_selector, seastar::reactor_config): Assertion `r >= 0' failed.
scylla: ../../src/core/reactor.cc:1297: seastar::reactor::reactor(unsigned int, seastar::reactor_backend_selector, seastar::reactor_config): Assertion `r >= 0' failed.
scylla: ../../src/core/reactor.cc:1297: seastar::reactor::reactor(unsigned int, seastar::reactor_backend_selector, seastar::reactor_config): Assertion `r >= 0' failed.
scylla: ../../src/core/reactor.cc:1297: seastar::reactor::reactor(unsigned int, seastar::reactor_backend_selector, seastar::reactor_config): Assertion `r >= 0' failed.
scylla: ../../src/core/reactor.cc:1297: seastar::reactor::reactor(unsigned int, seastar::reactor_backend_selector, seastar::reactor_config): Assertion `r >= 0' failed.
scylla: ../../src/core/reactor.cc:1297: seastar::reactor::reactor(unsigned int, seastar::reactor_backend_selector, seastar::reactor_config): Assertion `r >= 0' failed.
Aborting on shard 34.
Backtrace:
  0x00000000041a0bf2
  0x000000000408df05
  0x000000000408e205
  0x000000000408e2b3
  0x00007f562aecc02f
  /opt/scylladb/libreloc/libc.so.6+0x000000000003853e
  /opt/scylladb/libreloc/libc.so.6+0x0000000000022894
  /opt/scylladb/libreloc/libc.so.6+0x0000000000022768
  /opt/scylladb/libreloc/libc.so.6+0x00000000000309f5
  0x0000000004196ac6
  0x0000000004196db3
  0x000000000419705a
  0x00000000040596ad
  /opt/scylladb/libreloc/libpthread.so.0+0x000000000000858d
  /opt/scylladb/libreloc/libc.so.6+0x00000000000fd6a2

```

**Comment by tzach** [comment]
Did a quick test with Docker on Fedora and it works fine.
Need to retest with Debian 9

**Comment by tzach** [comment]
Tested with a fresh Debian 9 on Google Cloud Engine, still working as expected.

**Comment by rahulchaki** [comment]
Facing the same issue on Ubuntu 19.10. 
Installation details
Scylla version (or git commit hash): 3.1.0 (what I tried)
Cluster size: 1
OS (RHEL/CentOS/Ubuntu/AWS AMI): docker 19.03 on Ubuntu 19.10

Platform (physical/VM/cloud instance type/docker): docker 19.03

**Comment by Gleb-ko** [comment]
I am facing the same issue, same setup & installation as @rahulchaki .

**Comment by lawzava** [comment]
Identical issue on Ubuntu 19.10.
Scylla - from scylladb/scylla repo

**Comment by slivne** [comment]
- we have tried on gce on ubuntu 19/10 using scylla docker 3.1.0

```
root@sl-t-ubuntu19-10:/home/shlomi# docker --version
Docker version 19.03.6, build 369ce74a3c
root@sl-t-ubuntu19-10:/home/shlomi# 

```

```
root@sl-t-ubuntu19-10:/home/shlomi# cat /etc/os-release 
NAME="Ubuntu"
VERSION="19.10 (Eoan Ermine)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 19.10"
VERSION_ID="19.10"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=eoan
UBUNTU_CODENAME=eoan
```

we have run this on a 2 core machine.

There was an issue https://github.com/scylladb/scylla/issues/3842 - you can try and check if you are hitting this and change the fs.aio-max-nr and check if its resolved.

**Comment by slivne** [comment]
reproduced - changed the machine to be 32 core

```
2020-03-08 11:25:38,266 INFO exited: scylla (terminated by SIGABRT (core dumped); not expected)
2020-03-08 11:25:41,272 INFO spawned: 'scylla' with pid 257
scylla: ../../src/core/reactor.cc:1297: seastar::reactor::reactor(unsigned int, seastar::reactor_backend_selector, seastar::reactor_config): Assertion `r >= 0' failed.
Aborting on shard 8.
Backtrace:
  0x00000000041a0c82
  0x000000000408df95
  0x000000000408e295
  0x000000000408e343
  0x00007f66a246a02f
  /opt/scylladb/libreloc/libc.so.6+0x000000000003853e
  /opt/scylladb/libreloc/libc.so.6+0x0000000000022894
  /opt/scylladb/libreloc/libc.so.6+0x0000000000022768
  /opt/scylladb/libreloc/libc.so.6+0x00000000000309f5
  0x0000000004196b56
  0x0000000004196e43
  0x00000000041970ea
  0x000000000405973d
  /opt/scylladb/libreloc/libpthread.so.0+0x000000000000858d
  /opt/scylladb/libreloc/libc.so.6+0x00000000000fd6a2
2020-03-08 11:25:41,874 INFO exited: scylla (terminated by SIGABRT (core dumped); not expected)
2020-03-08 11:25:42,875 INFO gave up: scylla entered FATAL state, too many start retries too quickly
```

**Comment by slivne** [comment]
work around:
1. run scylla with limited smp
```
docker run --name some-scylla --hostname some-scylla -d scylladb/scylla:3.1.0 --smp 2
```
2.  set fs.aio-max-nr = 1048576

**Comment by slivne** [comment]
@avikivity / @tzach  maybe we should limit docker to run by default with smp=1 as the default is anyways to run in developer mode

**Comment by tzach** [comment]
> @avikivity / @tzach maybe we should limit docker to run by default with smp=1 as the default is anyways to run in developer mode

Make sense to me. One can always use SMP to change this.

**Comment by slivne** [comment]
@avikivity ping users are failing to run docker

**Comment by slivne** [comment]
this is fixed by da00530464779bb1d3e6757e2e804ca909ee0e5d

seastar patch scylladb/seastar@cc4115b9134e412a92f8f9ee52d7e395b9ca6c83

part of 3.3.0
